experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnInit } from "@angular/core";
+import { FormControl, FormGroup, Validators } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute, Router } from "@angular/router";
+import { ResponseStatus } from "trafficops-types";
+
+import { ServerService } from "src/app/api";
+import { DecisionDialogComponent, DecisionDialogData } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * StatusDetailsComponent is the controller for a status "details" page.
+ */
+@Component({
+	providers: [ServerService],
+	selector: "tp-status-details",
+	styleUrls: ["./status-details.component.scss"],
+	templateUrl: "./status-details.component.html",
+})
+export class StatusDetailsComponent implements OnInit {
+
+	/** Status ID expected from the route param using which we identify whether we are creating new status or load existing status */
+	public id: string | number | null = null;
+
+	/** All details of status requested */
+	public statusDetails: ResponseStatus | null = null;
+
+	/** Reactive form intialized to creat / edit status details */
+	public statusDetailsForm!: FormGroup;
```

Comment:
it'd be a lot simpler to just put the value here or in the constructor, instead of using not null assertion. The value doesn't depend on any binding to inputs or outputs of anything, so there's no reason to wait for `ngOnInit`